src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright 2024 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ReactElement, useCallback, useMemo, useState } from 'react'
+import { Modal as AntModal } from 'antd'
+import classNames from 'classnames'
+
+import { IconPosition, Shape, Type } from '../Button/Button.types'
+import { BodyM } from '../Typography/BodyX/BodyM'
+import { Button } from '../Button'
+import { H4 } from '../Typography/HX/H4'
+import { Icon } from '../Icon'
+import { ModalProps } from './Modal.props'
+import { Size } from './Modal.types'
+import styles from './Modal.module.css'
+import { useTheme } from '../../hooks/useTheme'
+
+const { Left, Right } = IconPosition
+const { Circle } = Shape
+const { Ghost } = Type
+const { Small, Large, FullScreen } = Size
+const {
+  body,
+  bodyFullWidth,
+  content,
+  contentWrapper,
+  extensionLabel,
+  footerButtons,
+  modal,
+  modalSm,
+  modalLg,
+  modalFs,
+} = styles
+
+/**
+ *
+ *
+ * @link https://ant.design/components/modal
+ * @returns {Modal} Modal component
+ */
+export const Modal = ({
+  children,
+  docLink,
```

Comment:
è un po' custom sul nostro uso in console come prop. mi torna la sua presenza però astrarrei il concetto.
Userei qualcosa tipo iconName che è una implementazione delle nostre icone e un iconLink?

Reply:
i agree to keep the docLink concept within the modal; we might add a more customizable link in the future but as of now it hardly makes sense to keep it outside the component